### DISCUSSION TITLE
Provisioner's `TransferProcess` automatic deprovisioning

### DISCUSSION THREAD

**Discussion by ndr-brt** [eclipse-edc/Connector#2542] [Ideas]
Currently the deprovisioning process is completely manual, this means that if no one takes care of it there could a potential lack of provisioned resources that aren't needed anymore.

While I get the need on the consumer side to let the data put in the provisioned resource (like a bucket or so) ready to be consumed, I don't see why the provider couldn't automatically deprovision the transfer resources as soon as it gets completed or terminated.

What do you think about?

**Comment by jimmarino** [comment]
Yes, this should be the case. Code used to perform deprovisioning a while back and the new state machine definitions should hopefully make this clear now when it needs to be done.
